testui/support/parallel.py

```diff
@@ -230,131 +296,152 @@ def __start_processes(markers: list, args, test_run_id=None):
+            if text.__contains__("ID="):
                 id_test = text.split("ID=")[1]
-                if text.split("ID=")[1].__contains__('\n'):
+                if text.split("ID=")[1].__contains__("\n"):
                     id_test = text.split("ID=")[1].split("\n")[0]
-                logger.log_info(f'Test run: {id_test}')
+                logger.log_info(f"Test run: {id_test}")
                 return id_test
-            raise Exception('Failed to create Test Run')
-        elif text.__contains__("Failed to create testrun"):
+            raise Exception("Failed to create Test Run")
+        if text.__contains__("Failed to create testrun"):
             out.close()
             process.send_signal(signal=2)
             process.terminate()
             process.wait()
-            os.remove('testrail_id_file.txt')
-            logger.log_error(f'Failed to create Test Run: \n {text}')
-            raise Exception(f'Failed to create Test Run')
+            os.remove("testrail_id_file.txt")
+            logger.log_error(f"Failed to create Test Run: \n {text}")
+            raise Exception("Failed to create Test Run")
         out.close()
         i += 1
 
 
 def __process(markers: list, args, thread=0, test_run_id=None):
-    for i, marker in enumerate(markers):
+    for marker in markers:
         try:
-            os.remove(f'.my_cache_dir_{thread}/v/cache/lastfailed')
-            os.remove(f'.my_cache_dir_{thread}/v/cache/nodeids')
-            os.remove(f'.my_cache_dir_{thread}/v/cache/stepwise')
+            os.remove(f".my_cache_dir_{thread}/v/cache/lastfailed")
+            os.remove(f".my_cache_dir_{thread}/v/cache/nodeids")
+            os.remove(f".my_cache_dir_{thread}/v/cache/stepwise")
         except FileNotFoundError:
             pass
-        quiet = '-q'
+        quiet = "-q"
         if args.s:
-            quiet = '-s'
-        testrail = ''
+            quiet = "-s"
+        testrail = ""
         if test_run_id is not None:
-            testrail = f'--testrail --tr-config=testrail.cfg --tr-run-id={test_run_id}'
+            testrail = (
+                f"--testrail --tr-config=testrail.cfg --tr-run-id={test_run_id}"
+            )
             if args.testrail_pwd is not None:
-                testrail += f' --tr-password={args.testrail_pwd}'
-        cache = f'-o cache_dir=.my_cache_dir_{thread}'
+                testrail += f" --tr-password={args.testrail_pwd}"
+        cache = f"-o cache_dir=.my_cache_dir_{thread}"
         start_1 = time.time()
-        logger.log(f'Starting: pytest {quiet} -m "{marker} {args.general_markers}" {testrail} {args.general} {cache}')
-        pr_1 = os.system(f'pytest {quiet} -m "{marker} {args.general_markers}" {testrail} {args.general} {cache}')
-        file = open('fails.txt', 'a+')
-        file.write(f'{pr_1}')
+        logger.log(
+            f'Starting: pytest {quiet} -m "{marker} {args.general_markers}" {testrail} {args.general} {cache}'
```

Comment:
I kind of would prefer that as-well, but the formatter "black" formats this automatically. From their doc:
```
Black prefers double quotes (" and """) over single quotes (' and '''). It will replace the latter with the former as long as it does not result in more backslash escapes than before.
```